flake/dev/generate-all-maintainers/extract-maintainers-meta.nix

```diff
@@ -0,0 +1,57 @@
+# Extract maintainers from nixvim configuration using meta.maintainers
+# This script evaluates an empty nixvimConfiguration and extracts the merged maintainer information
+let
+  nixvim = import ../../..;
+  pkgs = import <nixpkgs> { };
+  lib = pkgs.lib.extend nixvim.lib.overlay;
+
+  emptyConfig = lib.nixvim.evalNixvim {
+    modules = [ ];
+    extraSpecialArgs = { };
+    system = "x86_64-linux";
+  };
+
+  inherit (emptyConfig.config.meta) maintainers;
+
+  extractMaintainerObjects = maintainerData: lib.unique (lib.flatten (lib.attrValues maintainerData));
+
+  allMaintainerObjects = extractMaintainerObjects maintainers;
+
+  getMaintainerName =
+    maintainer:
+    if lib.hasAttr "github" maintainer then
+      maintainer.github
+    else if lib.hasAttr "name" maintainer then
+      maintainer.name
+    else
+      null;
+
+  allMaintainerNames = lib.filter (name: name != null) (map getMaintainerName allMaintainerObjects);
+
+  maintainerDetails = lib.listToAttrs (
+    map (obj: {
+      name = getMaintainerName obj;
+      value = obj;
+    }) (lib.filter (obj: getMaintainerName obj != null) allMaintainerObjects)
+  );
+  nixvimMaintainers = import ../../../lib/maintainers.nix;
+  nixvimMaintainerNames = lib.attrNames nixvimMaintainers;
+  categorizedMaintainers = {
+    nixvim = lib.filterAttrs (name: _: lib.elem name nixvimMaintainerNames) maintainerDetails;
+    nixpkgs = lib.filterAttrs (name: _: !(lib.elem name nixvimMaintainerNames)) maintainerDetails;
```

Comment:
This may also be possible with a `partition` or a `groupBy`. However, that'd require using `attrsToList` and `listToAttrs`.

If we cared about performance here, that'd mean only having a single pass checking the predicate. Here that shouldn't really matter, realistically.